### PR TITLE
[release/9.0.1xx-preview2] Update aotprofile 

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Update the AOTProfile

Runs before change
```
## Network stats: elapsed time=14ms (0ms mobile, 0ms wifi, 14ms not connected)
02-28 19:55:06.708  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +388ms
02-28 19:55:10.085  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +388ms
02-28 19:55:13.424  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +381ms
02-28 19:55:16.770  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +381ms
02-28 19:55:20.115  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +387ms
02-28 19:55:23.559  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +388ms
02-28 19:55:26.928  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +383ms
02-28 19:55:30.284  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +384ms
02-28 19:55:33.677  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +383ms
02-28 19:55:37.078  1618  1709 I ActivityTaskManager: Displayed com.companyname.foo1/crc6406c292f0e2f5d001.MainActivity: +377ms
Average(ms): 384
Std Err(ms): 1.18321595661992
Std Dev(ms): 3.74165738677394
```

Runs after change:

```
02-28 20:17:12.608  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +396ms
02-28 20:17:16.044  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +393ms
02-28 20:17:19.467  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +394ms
02-28 20:17:22.827  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +381ms
02-28 20:17:26.517  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +385ms
02-28 20:17:29.905  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +398ms
02-28 20:17:33.327  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +399ms
02-28 20:17:36.766  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +391ms
02-28 20:17:40.157  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +383ms
02-28 20:17:43.542  1618  1709 I ActivityTaskManager: Displayed com.companyname.foonew/crc6437c1864862c65670.MainActivity: +395ms
Average(ms): 391.5
Std Err(ms): 2.01246117974981
Std Dev(ms): 6.36396103067893
```

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
So the new profile got worse? Average(ms): 384 vs 391.5?

File: src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui.aotprofile.txt [hunk 1]
```diff
@@ -1661,19 +1657,15 @@ Methods:
...
 	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.AppThemeBinding:Clone ()
 	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.Binding:Clone ()
+	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.SetterSpecificityList`1<Microsoft.Maui.Controls.BindingBase>:get_Item (Microsoft.Maui.Controls.SetterSpecificity)
+	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.SetterSpecificityList`1<Microsoft.Maui.Controls.BindingBase>:GetClearedValue ()
+	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.SetterSpecificityList`1<Microsoft.Maui.Controls.BindingBase>:GetValue ()
 	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.Xaml.AppThemeBindingExtension:Microsoft.Maui.Controls.Xaml.IMarkupExtension<Microsoft.Maui.Controls.BindingBase>.ProvideValue (System.IServiceProvider)
-	Microsoft.Maui.Controls.BindingBase Microsoft.Maui.Controls.Xaml.BindingExtension:Microsoft.Maui.Controls.Xaml.IMarkupExtension<Microsoft.Maui.Controls.BindingBase>.ProvideValue (System.IServiceProvider)
```
Comment (jonathanpeppers) [inline src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui.aotprofile.txt:1669]: So we lost the built-in profile for bindings? https://github.com/dotnet/maui/commit/018c74670ee5e14abbd2141b13140affbfa6b701 says it saved 1ms:
```
Total (ms) |  Self (ms) | Method
          0.99 |       0.99 | Microsoft.Maui.Controls.Xaml.BindingExtension:Microsoft.Maui.Controls.Xaml.IMarkupExtension<Microsoft.Maui.Controls.BindingBase>.ProvideValue (System.IServiceProvider)
```
We could make a `ViewModel` and really bind the value? Or somehow ignore XC022?

File: src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt [hunk 2]
```diff
@@ -2698,8 +2803,11 @@ Methods:
 	System.Delegate System.Delegate:CreateDelegate (System.Type,System.Type,string)
 	System.Delegate System.Delegate:CreateDelegate_internal (System.Runtime.CompilerServices.QCallTypeHandle,object,System.Reflection.MethodInfo,bool)
 	System.Delegate System.Delegate:Remove (System.Delegate,System.Delegate)
+	System.Delegate System.Linq.Expressions.Compiler.LambdaCompiler:Compile (System.Linq.Expressions.LambdaExpression)
+	System.Delegate System.Linq.Expressions.Compiler.LambdaCompiler:CreateDelegate ()
```
Comment (jonathanpeppers) [inline src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt:2807]: This looks like something is compiling `System.Linq.Expressions` at startup. We should probably investigate what that is, as it's probably slowing down startup.

Comment (jonathanpeppers) [reply src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt:2807]: Oh this is only in the `maui-blazor` app, so it's probably something ASP.NET is doing. The `maui` app doesn't have `System.Linq.Expressions.Compiler`.

Comment (mattleibow) [reply src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt:2807]: That is new? Did ASP.NET start using LINQ in startup?

Comment (jonathanpeppers) [reply src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt:2807]: I've always seen stuff that wasn't great for startup on Android in the `maui-blazor` template. But it was always code in 
ASP.NET.

Comment (rmarinho) [reply src/Controls/src/Build.Tasks/nuget/buildTransitive/netstandard2.0/maui-blazor.aotprofile.txt:2807]: the numbers I posted were for maui and not maui-blazor.

**Review by jonathanpeppers — APPROVED** [review]
I think maybe @rmarinho, could you have been testing the same profile in both cases? I put comments about the NuGet cache in the `README.md`.

I restored the binding that was there before, but I think even without that I was seeing the new profile is better.

Latest I got on a Pixel 5:
```
Before:
02-29 08:37:48.821  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +616ms
02-29 08:37:50.586  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +604ms
02-29 08:37:52.353  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +604ms
02-29 08:37:54.135  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +598ms
02-29 08:37:55.959  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +628ms
02-29 08:37:57.806  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +657ms
02-29 08:37:59.613  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +640ms
02-29 08:38:01.488  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +668ms
02-29 08:38:03.277  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +610ms
02-29 08:38:05.075  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +611ms
Average(ms): 623.6
Std Err(ms): 7.60146184771207
Std Dev(ms): 24.0379329856421

After:
02-29 08:33:20.110  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +566ms
02-29 08:33:21.876  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +584ms
02-29 08:33:23.625  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +568ms
02-29 08:33:25.423  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +596ms
02-29 08:33:27.183  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +560ms
02-29 08:33:28.950  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +569ms
02-29 08:33:30.692  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +556ms
02-29 08:33:32.464  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +558ms
02-29 08:33:34.218  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +576ms
02-29 08:33:35.973  2053  2288 I ActivityTaskManager: Displayed com.companyname.hellomaui/crc640655b0e0c20d93c7.MainActivity for user 0: +573ms
Average(ms): 570.6
Std Err(ms): 3.90782690393409
Std Dev(ms): 12.3576337181157
```
So it seems like it got like 50ms better?

**Comment by rmarinho** [comment]
@jonathanpeppers  oh, didn't know about the cache, I didn't do a git clean or  anything only did a dotnet pack release after changing the profile as per instructions.

**Comment by jonathanpeppers** [comment]
@rmarinho after this is merged, can we put it on the `net9.0` branch as well?

That would should get better times for the pipeline that tracks startup at: https://aka.ms/dotnetperfstatus

**Pull request merged**